connector/src/main/java/org.apache.flink/connector/nebula/catalog/NebulaCatalog.java

```diff
@@ -93,11 +160,62 @@ public CatalogDatabase getDatabase(String databaseName) throws DatabaseNotExistE
         }
     }
 
+    /**
+     * @param s database name (same as graph space name)
+     * @param catalogDatabase catalog impl
+     * @param b true if contains [if not exists] else false
+     */
+    @Override
+    public void createDatabase(String s, CatalogDatabase catalogDatabase, boolean b)
+            throws CatalogException {
+        checkArgument(
+                !isNullOrWhitespaceOnly(s), "space name cannot be null or empty.");
+        checkNotNull(catalogDatabase, "space cannot be null.");
+
+        if (b && listDatabases().contains(s)) {
+            LOG.error("failed to create graph space {}, already exists", s);
+            throw new CatalogException("nebula create graph space failed.");
+        }
+        Map<String, String> properties = catalogDatabase.getProperties();
+        Map<String, String> newProperties = properties.entrySet().stream().collect(
+            Collectors.toMap(
+                entry -> entry.getKey().toLowerCase(),
+                entry -> entry.getValue().toUpperCase()
+            )
+        );
+        if (!newProperties.containsKey(NebulaConstant.CREATE_VID_TYPE)) {
+            LOG.error("failed to create graph space {}, missing VID type param", properties);
+            throw new CatalogException("nebula create graph space failed, missing VID type.");
+        }
+        String vidType = newProperties.get(NebulaConstant.CREATE_VID_TYPE);
+        if (!NebulaUtils.checkValidVidType(vidType)) {
+            LOG.error("VID type not satisfy {}", vidType);
+            throw new CatalogException("nebula graph dont support VID type.");
+        }
+        NebulaSpace space = new NebulaSpace(s, catalogDatabase.getComment(), newProperties);
+        NebulaSpaces nebulaSpaces = new NebulaSpaces(space);
+        String statement = nebulaSpaces.getCreateStatement();
+        ResultSet execResult = null;
+        try {
+            execResult = session.execute(statement);
+        } catch (IOErrorException e) {
+            LOG.error("nebula create graph space execute failed.", e);
+            throw new CatalogException("nebula create graph space execute failed.");
+        }
+
+        if (execResult.isSucceeded()) {
+            LOG.debug("create space success.");
+        } else {
+            LOG.error("create space failed: {}", execResult.getErrorMessage());
+            throw new CatalogException("create space failed.");
```

Comment:
append the error message of execResult

Reply:
it means to append execResult error message to CatalogException message?

Reply:
yeah

Reply:
got it, [e23cdfd](https://github.com/vesoft-inc/nebula-flink-connector/pull/66/commits/e23cdfdce509bd9bf35ce67b4a4b04054b5de00d) should sovle this i think, thanks for your kindly suggestion.